src/Hosting/Server.IntegrationTesting/src/ApplicationPublisher.cs

```diff
@@ -38,7 +38,7 @@ public virtual Task<PublishedApplication> Publish(DeploymentParameters deploymen
                                  // avoids triggering builds of dependencies of the test app which could cause issues like https://github.com/dotnet/arcade/issues/2941
                                  + $" --no-dependencies"
                                  + $" /p:TargetArchitecture={deploymentParameters.RuntimeArchitecture}"
-                                 + " --no-restore";
+                                 + (deploymentParameters.RestoreDependencies ? "" : " --no-restore");
```

Comment:
I recall no-restore is required for all tests on Helix because the test assets need to be pre-built before publishing to helix.

Reply:
Works fine on helix as far as I could tell.